test/rubygems/test_gem_ext_rake_builder.rb

```diff
@@ -24,7 +24,7 @@ def test_class_build
 
       refute_match %r{^rake failed:}, output
       assert_match %r{^#{Regexp.escape Gem.ruby} mkrf_conf\.rb}, output
-      assert_match %r{^#{Regexp.escape rake} RUBYARCHDIR\\=#{Regexp.escape @dest_path} RUBYLIBDIR\\=#{Regexp.escape @dest_path}}, output
+      assert_match %r{^#{Regexp.escape rake} RUBYARCHDIR=#{Regexp.escape @dest_path} RUBYLIBDIR=#{Regexp.escape @dest_path}}, output
```

Comment:
This can be actually considered a bug fix.

The previous shellscaping of arguments was meant for arguments to "survive" the extra shell and arrive with the correct value to the final program (`rake` in this case). But for error messages, in addition to making the command less clear, copy-pasting it into your shell won't work.

For example, in this case, `rake`'s behavior when given `=` separated args is to set environment variables. See https://github.com/ruby/rake/blob/c2eeae2fe2b67170472a1441ebf84d3a238c3361/lib/rake/application.rb#L751-L759.

Using a "shellescaped" command breaks that:

```
$ echo "puts ENV['RUBYLIBDIR']" > Rakefile
$ rake RUBYLIBDIR=. -T
.
$ rake RUBYLIBDIR\\=. -T
```

Reply:
I decided to be a good boy and splitted what I think is a bug fix to a separate PR: https://github.com/rubygems/rubygems/pull/4194.

Reply:
Quoting whitespaces may be needed instead?

Reply:
As far as I understand, quoting wouldn't be needed since there's no longer a shell involved? In any case, we're not currently quoting anything, so I think it'd be part of a separate bug fix if needed?

Reply:
> But for error messages, in addition to making the command less clear, copy-pasting it into your shell won't work.

When copy-pasting, spaces in a "word" need to be quoted or escape, no?

Reply:
Oh, you're totally correct @nobu. I was thinking of the command that gets run, not the command that we print. So... I'll have yet another look.

Reply:
It should be fixed now!